File: GameDev/JsonData.cs
using System;
using System.IO;
using Newtonsoft.Json.Linq;

namespace GameDev
{
    public class JsonData
    {
        public String filename = "";
        public JToken data = null;        

        public void Clear()
        {
            filename = "";
            data = null;
        }

        public void Load()
        {
            data = JToken.Parse(File.ReadAllText(filename));
        }

        public void Save()
        {
            File.WriteAllText(filename, data.ToString());
        }

    }
}

File: QGameDev/JsonData.cpp
#include <QFile>
#include <QJsonDocument>
#include "JsonData.h"

void JsonData::Clear()
{
	filename = "";
	data = QJsonDocument();
}

void JsonData::Load()
{
	QFile file;
	file.setFileName(filename);
	file.open(QIODevice::ReadOnly | QIODevice::Text);
	QString text = file.readAll();
	file.close();
	data = QJsonDocument::fromJson(text.toUtf8());	
}

void JsonData::Save()
{
	QFile file;
	file.setFileName(filename);
	file.open(QFile::WriteOnly);
	file.write(data.toJson());
	file.close();
}

